### PR TITLE
State filter

### LINKED ISSUE #1128: State Filter
Allow users to filter by state: 
things that need to be done for this to be completed: 

- [ ] filter design
- [ ] filter functionality
- [ ]  parameter support

### LINKED ISSUE #1129: State filter design

### PR DESCRIPTION
**Pull request by edwingl10**
## Motivation and Context

Fixes #1128 

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] User visible change (users will notice UI or functional changes)

## How Has This Been Tested?

- [ ] Existing or new tests cover my changes.
- [ ] Manually tested locally or on the Render PR Server.

## Previewing Changes

1. Click the `onrender.com` URL the **render `bot`** posted below.
2. go to scholarships page 
3. add state filters

### CODE REVIEW

File: src/components/CustomAutocomplete.tsx [hunk 1]
```diff
@@ -28,7 +28,8 @@ export default function AutocompleteFilter(
       open={isOpen && !limitReached}
       disabled={limitReached}
       onInputChange={(e, val) => setIsOpen(val.length > 0)}
-      renderTags={() => null}
+      forcePopupIcon={false}
```
Comment (edwingl10) [inline src/components/CustomAutocomplete.tsx:31]: helps with the previous bug I had.
Possible [explanation.](https://stackoverflow.com/questions/61256685/material-ui-autocomplete-styling)

File: src/components/CustomAutocomplete.tsx [hunk 2]
```diff
@@ -51,12 +52,12 @@ export default function AutocompleteFilter(
...
+CustomAutocomplete.propTypes = {
   limitReached: PropTypes.bool,
   placeholder: PropTypes.string,
 };
 
-AutocompleteFilter.defaultProps = {
+CustomAutocomplete.defaultProps = {
   limitReached: false,
   placeholder: undefined,
 };
```
Comment (josueetcom) [inline src/components/CustomAutocomplete.tsx:63]: Why does `CustomAutoCompete` have propTypes and defaultProps?

File: src/components/FilterPanel.jsx [hunk 3]
```diff
@@ -50,15 +53,16 @@ export default function FilterPanel({ onClose }) {
...
 
   const filters = {
     'What are you studying?': {
       comp: (
         <>
-          <AutocompleteFilter
+          <CustomAutocomplete
             freeSolo
-            value={majors}
+            value={majors || []}
```
Comment (josueetcom) [inline src/components/FilterPanel.jsx:65]: Why is this necessary? Also why not have the `|| []` logic in the component itself?

File: src/lib/useQueryParams.ts [hunk 4]
```diff
@@ -95,6 +96,19 @@ export default function useQueryParams(
         delete params.majors;
       }
     }
+
+    if (states !== undefined) {
```
Comment (josueetcom) [inline src/lib/useQueryParams.ts:100]: You need to update the tests

File: src/models/Scholarships.ts [hunk 5]
```diff
@@ -97,6 +97,7 @@ export interface FilterOptions {
   maxAmount?: number;
   grades?: GradeLevel[];
   majors?: String[];
```
Comment (josueetcom) [inline src/models/Scholarships.ts:99]: Tangential but can you lower-case the `s` here for consistency?
```suggestion
  majors?: string[];
```

File: src/components/CustomAutocomplete.tsx (hunk 1 above)
Comment (josueetcom) [reply src/components/CustomAutocomplete.tsx:31]: What does `forcePopupIcon` do? I understand it does something with the up/down icon on the right but I couldn't make heads or tail of the API description when it says "Force the visibility display"

File: src/components/CustomAutocomplete.tsx (hunk 2 above)
Comment (edwingl10) [reply src/components/CustomAutocomplete.tsx:63]: yeah I guess it doesn't need them.

File: src/components/FilterPanel.jsx (hunk 3 above)
Comment (edwingl10) [reply src/components/FilterPanel.jsx:65]: because it was complaining about the following: `MUI: A component is changing the uncontrolled value state of Autocomplete to be controlled.`

Comment (edwingl10) [reply src/components/FilterPanel.jsx:65]: I tried doing it in the component itself but interestingly, it wouldn't let me type in the state filter. It's the same bug that we have in the scholarship form

**Review by josueetcom — APPROVED** [review]

File: src/components/CustomAutocomplete.tsx (hunk 1 above)
Comment (edwingl10) [reply src/components/CustomAutocomplete.tsx:30]: I think besides visibility, the styles are also not forced in this case.
